### PR TITLE
TokenLogic & karpatkey - Service Providers

### PR DESCRIPTION
**Pull request by efecarranza**
## TODO: Pending Snapshot

# Changelog

Add proposal to add TokenLogic and karpatkety as service providers by creating a stream.
Add tests for streams.
Add MD file with AIP.

### CODE REVIEW

File: src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership_20231207.sol [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+pragma solidity 0.8.19;
+
+import {IProposalGenericExecutor} from 'aave-helpers/interfaces/IProposalGenericExecutor.sol';
+import {AaveV3Ethereum, AaveV3EthereumAssets} from 'aave-address-book/AaveV3Ethereum.sol';
+
+/**
+ * @title TokenLogic & karpatkey Service Provider Partnership
+ * @author efecarranza.eth
+ * - Snapshot: TODO
+ * - Discussion: https://governance.aave.com/t/temp-check-financial-services-proposal-karpatkey-tokenlogic/15633
```
Comment (eboadom) [inline src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership_20231207.sol:12]: Link should be to ARFC, to be consistent on for example the assets used (GHO on ARFC, vs others in TEMP CHECK)

File: src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership_20231207.t.sol [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+   */
+  function test_execute() public {
+    (, , , , , , uint256 streamToCancelRemaining, ) = AaveV3Ethereum.COLLECTOR.getStream(
+      proposal.STREAM_TO_CANCEL()
+    );
+
+    assertGt(streamToCancelRemaining, 0);
+
+    vm.startPrank(proposal.STREAM_TWO_RECEIVER());
+    AaveV3Ethereum.COLLECTOR.withdrawFromStream(
```
Comment (eboadom) [inline src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership_20231207.t.sol:36]: For what is this?

Comment (efecarranza) [reply src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership_20231207.t.sol:36]: Cancel existing stream, to test that the available funds are moved.

File: src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership_20231207.sol (hunk 1 above)
Comment (efecarranza) [reply src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership_20231207.sol]: Done.

File: src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership_20231207.t.sol (hunk 2 above)
Comment (Matt-Graham-TL) [reply src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership_20231207.t.sol:36]: Reference: https://governance.aave.com/t/arfc-tokenlogic-hohmann-transfer/15051

File: src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/TokenLogicKarpatkeyServiceProviderPartnership.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
+---
+title: "TokenLogic & karpatkey Service Provider Partnership"
+author: "TokenLogic & karpatkey"
+discussions: "https://governance.aave.com/t/temp-check-financial-services-proposal-karpatkey-tokenlogic/15633"
```
Comment (eboadom) [inline src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/TokenLogicKarpatkeyServiceProviderPartnership.md:4]:
```suggestion
discussions: "https://governance.aave.com/t/arfc-tokenlogic-karpatkey-service-provider-partnership/15755"
```

File: src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership_20231207.t.sol (hunk 2 above)
Comment (eboadom) [reply src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership_20231207.t.sol:36]: Was meaning about the need to test this, but it is fine. Resolved

File: src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/TokenLogicKarpatkeyServiceProviderPartnership.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+# Disclaimer
+
+TokenLogic and karpatkey receive no compensation beyond Aave protocol for the creation of this proposal. TokenLogic and karpatkey are both delegates within the Aave ecosystem.
+
+## References
+
+- Implementation: [AaveV3Ethereum](https://github.com/bgd-labs/aave-proposals-v3/blob/main/src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership_20231207.sol)
+- Tests: [AaveV3Ethereum](https://github.com/bgd-labs/aave-proposals-v3/blob/main/src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership_20231207.t.sol)
+- [Snapshot](TODO)
```
Comment (sakulstra) [inline src/20231207_AaveV3Ethereum_TokenLogicKarpatkeyServiceProviderPartnership/TokenLogicKarpatkeyServiceProviderPartnership.md:105]:
```suggestion
- [Snapshot](https://snapshot.org/#/aave.eth/proposal/0x6e57ed54cf803652ab4839ff7b7f7de08ba086fbe99163547c6188a3ee55e209)
```

**Pull request merged**